tests/integration/test_authd/test_remote_enrollment.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import pytest
+from contextlib import nullcontext as does_not_raise
+
+from wazuh_testing.tools import monitoring, LOG_FILE_PATH
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.monitoring import SocketController, FileMonitor
+from wazuh_testing.tools.sockets import wait_for_tcp_port
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+
+# Configurations
+
+parameters = [
+    {'REMOTE_ENROLLMENT': 'no', 'CLUSTER_DISABLED': 'yes'},
+    {'REMOTE_ENROLLMENT': 'yes', 'CLUSTER_DISABLED': 'yes'},
+    {'REMOTE_ENROLLMENT': 'no', 'CLUSTER_DISABLED': 'no', 'NODE_TYPE': 'master'},
+    {'REMOTE_ENROLLMENT': 'yes', 'CLUSTER_DISABLED': 'no', 'NODE_TYPE': 'master'},
+    {'REMOTE_ENROLLMENT': 'no', 'CLUSTER_DISABLED': 'no', 'NODE_TYPE': 'worker'},
+    # x{'REMOTE_ENROLLMENT': 'yes', 'CLUSTER_DISABLED': 'no', 'NODE_TYPE': 'worker'},
+]
+
+metadata = [
+    {'remote_enrollment': 'no', 'id': 'no_remote_enrollment_standalone'},
+    {'remote_enrollment': 'yes', 'id': 'yes_remote_enrollment_standalone'},
+    {'remote_enrollment': 'no', 'id': 'no_remote_enrollment_cluster_master'},
+    {'remote_enrollment': 'yes', 'id': 'yes_remote_enrollment_cluster_master'},
+    {'remote_enrollment': 'no', 'id': 'no_remote_enrollment_cluster_worker'},
+
+    # {'remote_enrollment': 'yes', 'id': 'yes_remote_enrollment_cluster_worker}
+    # this fails with 'ERROR: Cannot communicate with master', would require a master node to work
+]
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf.yaml')
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+
+# Variables
+log_monitor_paths = []
+
+receiver_sockets_params = [(("localhost", 1515), 'AF_INET', 'SSL_TLSv1_2')]
```

Comment:
It would be interesting to change the port to a different one. Just in case.

Reply:
https://github.com/wazuh/wazuh-qa/pull/1368/commits/a283be84da01e3d068feb6651acd9c5af6b1606f